### ISSUE TITLE
popTime fails when given a data.table

### ISSUE THREAD

**Issue by sahirbhatnagar** [sahirbhatnagar/casebase#76]
I don't know why `casebase::popTime` fails when given a `data.table`, but it might be related to [FAQ 1.1](https://cran.r-project.org/web/packages/data.table/vignettes/datatable-faq.html#j-num). Something changed Nov 2016. It's something happening in `casebase:::checkArgsEventIndicator`

``` r
library(casebase)
#> See example usage at http://sahirbhatnagar.com/casebase/
popTimeData <- popTime(data = bmtcrr, time = "ftime", event = "Status")

bt <- data.table::as.data.table(bmtcrr)
popTimeData <- popTime(data = bt, time = "ftime", event = "Status")
#> Error in `[.data.table`(data, , event): j (the 2nd argument inside [...]) is a single symbol but column name 'event' is not found. Perhaps you intended DT[, ..event]. This difference to data.frame is deliberate and explained in FAQ 1.1.
```

<sup>Created on 2020-02-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by sahirbhatnagar** [comment]
The problem I think is in [`data[,event]`](https://github.com/sahirbhatnagar/casebase/blob/master/R/utils.R#L98)  in `casebase:::checkArgsEventIndicator`. This should be changed to `data[[event]]`

I'll make the change in my next PR

**Comment by turgeonmaxime** [comment]
We had similar issues earlier on with the other functions; see issue #5 

If you look at the code, you can still see a few places where I left the old code commented out; see for example [these lines](https://github.com/sahirbhatnagar/casebase/blob/24359bef3d070d52d558b5ba89956a81b71af898/R/sampling.R#L83-L84).

**Comment by turgeonmaxime** [comment]
If/when you fix these lines, make sure to add some tests!

**Comment by sahirbhatnagar** [comment]
This was closed by PR #80
